src/main/java/bio/terra/janitor/service/cleanup/flight/AzureNetworkSecurityGroupCleanupStep.java

```diff
@@ -0,0 +1,56 @@
+package bio.terra.janitor.service.cleanup.flight;
+
+import bio.terra.janitor.app.configuration.CrlConfiguration;
+import bio.terra.janitor.db.JanitorDao;
+import bio.terra.janitor.db.ResourceMetadata;
+import bio.terra.janitor.generated.model.AzureNetworkSecurityGroup;
+import bio.terra.janitor.generated.model.CloudResourceUid;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.compute.ComputeManager;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Step to clean up an Azure network security group. */
+public class AzureNetworkSecurityGroupCleanupStep extends ResourceCleanupStep {
+  private static final Logger logger =
+      LoggerFactory.getLogger(AzureNetworkSecurityGroupCleanupStep.class);
+  private final CrlConfiguration crlConfiguration;
+
+  public AzureNetworkSecurityGroupCleanupStep(
+      CrlConfiguration crlConfiguration, JanitorDao janitorDao) {
+    super(janitorDao);
+    this.crlConfiguration = crlConfiguration;
+  }
+
+  @Override
+  protected StepResult cleanUp(CloudResourceUid resourceUid, ResourceMetadata metadata)
+      throws InterruptedException, RetryException {
+    AzureNetworkSecurityGroup networkSg = resourceUid.getAzureNetworkSecurityGroup();
+    ComputeManager computeManager =
+        crlConfiguration.buildComputeManager(networkSg.getResourceGroup());
+
+    try {
```

Comment:
optional: We might be able to exact a common method that execute a method and return a step? 
Example:
https://github.com/DataBiosphere/terra-resource-buffer/blob/master/src/main/java/bio/terra/buffer/service/resource/flight/GoogleUtils.java#L100-L109

Reply:
Yeah.. I tried to do something similar at first but there are some slight differences and it was a bit of a pain. Maybe will revisit though since there is some duplicated code.

Reply:
We might be able to do something similar to the example above.
1: Define a FunctionalInterface 
```
  @FunctionalInterface
  public interface StairwayExecute {
    void execute();
  }
```
Then a util method: 
```
  public static StepResult DeleteAzureResourceAndIgnoreNotFound(StairwayExecute execute) {
    try {
      execute.execute();
    } catch (ManagementException e) {
            if (StringUtils.equals(e.getValue().getCode(), "ResourceNotFound")) {
        logger.info(
            "Azure network security group {} in managed resource group {} already deleted",
            networkSg.getNetworkSecurityGroupName(),
            networkSg.getResourceGroup().getResourceGroupName());
        return StepResult.getStepResultSuccess();
    }
  }
```

Reply:
Pushed up a change to do this :)